doc/rfc-move-sigverify-out-of-mask-rom.md

```diff
@@ -0,0 +1,80 @@
+# RFC: Move sigverify out of the mask ROM
+
+Author: James Wainwright
+Date: 2025-04-17
+Status: draft
+
+## Summary
+
+This RFC proposes moving the silicon creator signature verification code from the mask ROM to the immutable ROM extension, which would no longer be optional nor signed.
+
+The mutable ROM extension would be renamed to the FMC (first mutable code), and the term ROM extension would refer only to what we currently call the immutable ROM extension.
+
+## Motivation
+
+Moving sigverify out of the mask ROM enables:
+
+* Fixing and updating the cryptography code post tape-out.
+* Using different signature schemes for different SKUs of the same tape-out.
+
+Signature verification is relatively complex code and removing it from the mask derisks tape-outs.
+
+Allowing for personalised signature schemes increases the utility and value of a tape-out and potentially reduces costs if multiple customers can benefit. It also allows for SKUs which don't use signature verification and are closer to a generic microcontroller.
+
+Experimental and new signature schemes (for example using post-quantum cryptography) can be tested on a subset of personalised silicon without needing to be included in all chips.
+
+Simultaneously, making the immutable ROM extension non-optional simplifies the boot flow by keeping it linear.
```

Comment:
I think it'd be good to explain the expected configuration. I know Earlygrey is mentioned below, but I think it's important to call out assumptions such as "chips with embedded flash that is XIP".

Reply:
Good point, I've added this assumption to the explanation with a note about adapting it for external or non-XIP flash in the future work section.